rfcs/react-components/convergence/motion-definition-n-apis.md

```diff
@@ -0,0 +1,999 @@
+
+```tsx
+import { createAtom, atoms } from '@fluentui/react-motions-preview';
+
+const FadeEnterSlow = createAtom(atoms.fade.enterSlow);
+
+function MyComponent() {
+  return (
+    <FadeEnterSlow iterations={Infinity}>
+      <div>Hello world!</div>
+    </FadeEnterSlow>
+  );
+}
+```
+
+> Note: `createAtom()` returns a React component that clones the child element and applies the provided animation to it.
+
+<details>
+  <summary>Optional utility components?</summary>
+
+We can also expose a set of pre-defined components, for example:
+
+```tsx
+import { Collapse } from '@fluentui/react-motions-preview';
+
+function MyComponent() {
+  return (
+    <Collapse>
+      <div>Hello world!</div>
+    </Collapse>
+  );
+}
+```
+
+However, as we need to expose factories anyway, the proposal is to expose them as the primary API.
+
+</details>
+
+<details>
+  <summary>Why not follow `styled-components` / `framer-motion` API?</summary>
+
+Another API approach might be to compose existing components with motion factories, for example:
+
+```tsx
+// ⚠️ This is not proposed API, it's just an example
+
+import { Button } from '@fluentui/react-components';
+import { createAtomElement, atom } from '@fluentui/react-motions-preview';
+
+const FadeEnterDiv = createAtomElement('div', atom.fade.enterSlow);
+const FadeEnterButton = createAtomElement(Button, atom.fade.enterSlow);
+
+function MyComponent() {
+  return (
+    <>
+      <FadeEnterDiv>Hello world!</FadeEnterDiv>
+      <FadeEnterButton>Hello world!</FadeEnterButton>
+    </>
+  );
+}
+```
+
+This approach does not have obvious pros or cons, but it's worth noting that every animated component will have to be composed, for example:
+
+```tsx
+// ⚠️ This is not proposed API, it's just an example
+
+import { Button, Image } from '@fluentui/react-components';
+import { createAtom, createAtomElement, atom } from '@fluentui/react-motions-preview';
+
+const FadeEnter = createAtom(atom.fade.enterSlow);
+
+const FadeEnterButton = createAtomElement(Button, atom.fade.enterSlow);
+const FadeEnterImage = createAtomElement(Image, atom.fade.enterSlow);
+
+function MyComponent() {
+  return (
+    <>
+      <FadeEnter>
+        <div>Hello world!</div>
+      </FadeEnter>
+      <FadeEnter>
+        <Button>Hello world!</Button>
+      </FadeEnter>
+
+      {/* vs */}
+
+      <FadeEnterDiv>Hello world!</FadeEnterDiv>
+      <FadeEnterButton>Hello world!</FadeEnterButton>
+    </>
+  );
+}
+```
+
+</details>
+
+### Using presence motions
+
+Compared to the CSS option, we don't need to apply classes to the element, so factories don't need to control state and won't suffer from the same issues as the CSS option with re-rendering.
+
+> The Web Animations API has its own state machine [by design](https://developer.mozilla.org/en-US/docs/Web/API/Animation/playState), so we can subscribe, for example, to the onfinish event to handle unmount.
+
+<details>
+<summary>Handling unmount using callbacks</summary>
```

Comment:
It's not very from this example how the animation events relate to the presence components that are created by this API

I think there should be some way to link the unmount handling to `createPresence` since that is essentially how a presence handles unmounting on exit.

Perhaps you could reduce both details blocks to one single details block?

Reply:
I tried to improve this in c4a74dd. Does it make sense now or should I just remove this block?